content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,60 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+A NetworkPolicy is applied to a selected pod, when isolation rules are enforced
+for that pod. At that point pod can never establish an outbound connection or receive
+an inbound connection that is not allowed by the NetworkPolicy. Allow rules will be applied
+eventually after the isolation rules (that includes applying both at the same time).
+That means, that once the NetworkPolicy is applied to a pod, the pod should be able to handle situations,
+when it can't establish a connection that is allowed by an existing NetworkPolicy for some time.
+
+Every NetworkPolicy will be applied to all selected pods eventually, but there is no
+way to tell from the existing Kubernetes API when exactly the network policy is applied to a specific
+pod. The only way to check if the network policy is applied to a pod, is to run a connectivity check.
+
+Network plugins may implement extra `status.conditions` for pod readiness based on the
+status of network policies. These conditions then may be used by 
+[Pod readiness gate](/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate).
+
+## NetworkPolicy and `hostNetwork` pods
+
+NetworkPolicy behaviour for `hostNetwork` pods is undefined, when
+1. a `hostNetwork` pod is selected by `spec.podSelector`.
+   
+   For example, if there is a `hostNetwork` pod on the nodeA (pod IP equals nodeA IP in that case), 
+   that has label `role: client`, the following NetworkPolicy:
+   ```yaml
+     ...
+     spec:
+       podSelector:
+         matchLabels:
+           role: client
+     ...
+   ```
+   may apply ingress and egress rules to all connections with nodeA IP, or just ignore
+   that pod as if it didn't match the selector, or do something else.
```

Comment:
no, it is unambiguously wrong to apply the networkpolicy to nodeA IP traffic that _isn't_ to/from the hostNetwork pod. You have to either apply the policy _only_ to the hostNetwork traffic, or not apply it to any nodeA IP traffic

Reply:
I tried to just explain what undefined behaviours may be, since i guess we agreed on it here https://github.com/kubernetes/website/pull/39875#discussion_r1168498073
I can add one more sentence saying that most plugins will ignore hostNetwork pods, or do you prefer some other option?

Reply:
"Undefined" here doesn't mean that the plugin can do anything it wants. There are only two permitted possibilities:
1. The plugin implements the policy "correctly", by applying it to traffic to/from the hostNetwork pod, but _not_ applying it to other traffic with the node IP. Most plugins are not able to do this though.
2. The plugin ignores `hostNetwork` pods for purposes of matching `podSelector`s. (This is the expected behavior, though that's not documented.)

The plugin _can't_ "apply ingress and egress rules to all connections with nodeA IP". That's just wrong.